doc/system_overview.txt

```diff
@@ -0,0 +1,26 @@
+@startuml
```

Comment:
add a makefile which generates output for this in case we need to run it in future?